src/docfx/serve/lsp/credentialRefresh/CredentialRefreshParams.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+using MediatR;
+using OmniSharp.Extensions.JsonRpc;
+
+namespace Microsoft.Docs.Build
+{
+    [Parallel]
+    [Method("docfx/userCredentialRefresh", Direction.ServerToClient)]
+    public record CredentialRefreshParams : IRequest<CredentialRefreshResponse>
+    {
+        public CredentialType Type { get; init; }
```

Comment:
Instead of using a enum of token types, consider use URLs and HTTP standard response headers at the protocol layer (They are both standard), if we have other credentials in the future, the protocol does not need to change.
```suggestion
        public string BaseUrl { get; init; }
```